test/ha/workload_test.go

```diff
@@ -0,0 +1,171 @@
+			autoscaling.TargetBurstCapacityKey: "-1", // Make sure all requests go through the activator.
+		}),
+	)
+	test.EnsureTearDown(t, clients, &names)
+
+	testUptimeDuringUserPodDeletion(t, ctx, clients, names, resources)
+}
+
+func TestActivatorInRequestPathPossibly(t *testing.T) {
+	clients := e2e.Setup(t)
+	ctx := context.Background()
+
+	// Create first service that we will continually probe during disruption scenario.
+	names, resources := createPizzaPlanetService(t,
+		rtesting.WithConfigAnnotations(map[string]string{
+			autoscaling.MinScaleAnnotationKey:  strconv.Itoa(minimumNumberOfReplicas), // Make sure we don't scale to zero during the test.
+			autoscaling.MaxScaleAnnotationKey:  strconv.Itoa(maximumNumberOfReplicas),
+			autoscaling.TargetBurstCapacityKey: "1", // The Activator may be in the path, depending on the revision scale and load.
+		}),
+	)
+	test.EnsureTearDown(t, clients, &names)
+
+	testUptimeDuringUserPodDeletion(t, ctx, clients, names, resources)
+}
+
+func testUptimeDuringUserPodDeletion(t *testing.T, ctx context.Context, clients *test.Clients, names test.ResourceNames, resources *v1test.ResourceObjects) {
+	t.Log("Starting prober")
+	prober := test.NewProberManager(t.Logf, clients, minProbes, test.AddRootCAtoTransport(context.Background(), t.Logf, clients, test.ServingFlags.HTTPS))
+
+	prober.Spawn(resources.Service.Status.URL.URL())
+	defer assertSLO(t, prober, 1)
+
+	// Get user pods
+
+	selector := labels.SelectorFromSet(labels.Set{
+		serving.ServiceLabelKey: names.Service,
+	})
+	pods, err := clients.KubeClient.CoreV1().Pods(test.ServingFlags.TestNamespace).List(ctx, metav1.ListOptions{LabelSelector: selector.String()})
+	if err != nil {
+		t.Fatalf("Unable to get pods: %v", err)
+	}
+
+	if !(len(pods.Items) == minimumNumberOfReplicas) {
+		t.Fatalf("Expected to have %d user pod(s) running, but found %d.", minimumNumberOfReplicas, len(pods.Items))
+	}
+
+	t.Logf("Watching user pods")
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go watchPodEvents(t, ctx, clients, &wg, selector, pods.Items[0].Name)
+	go watchPodEvents(t, ctx, clients, &wg, selector, pods.Items[1].Name)
+
+	// Delete user pods
+	t.Logf("Deleting user pods")
+	for _, pod := range pods.Items {
+		err := clients.KubeClient.CoreV1().Pods(test.ServingFlags.TestNamespace).Delete(ctx, pod.Name, metav1.DeleteOptions{})
+		if err != nil {
+			t.Fatalf("Unable to delete pod: %v", err)
+		}
+	}
+
+	wg.Wait()
+
+	newPods, err := clients.KubeClient.CoreV1().Pods(test.ServingFlags.TestNamespace).List(ctx, metav1.ListOptions{LabelSelector: selector.String()})
+	if err != nil {
+		t.Fatalf("Unable to get pods: %v", err)
+	}
+
+	if !(len(newPods.Items) == minimumNumberOfReplicas) {
+		t.Errorf("Expected to have %d user pod(s) running, but found %d.", minimumNumberOfReplicas, len(newPods.Items))
+	}
+}
+
+func watchPodEvents(t *testing.T, ctx context.Context, clients *test.Clients, wg *sync.WaitGroup, selector labels.Selector, targetPod string) {
```

Comment:
In test pkg we have a func `WaitForPodDeleted`, would that work instead?

Reply:
oh cool, I'll check it out this func. Thanks!

Reply:
Replaced watchPodEvents with WaitForPodDeleted